src/applications/vaos/tests/new-appointment/components/ReasonForAppointmentPage.unit.spec.jsx

```diff
@@ -1,189 +1,236 @@
-    const routeToNextAppointmentPage = sinon.spy();
+  it('should show error msg when enter all spaces for Community Care medical request', async () => {
+    const store = createTestStore(initialState);
+    await setTypeOfFacility(store, /Community Care/i);
+
+    const screen = renderWithStoreAndRouter(<ReasonForAppointmentPage />, {
+      store,
+    });
 
-    const form = mount(
-      <ReasonForAppointmentPage
-        openReasonForAppointment={openReasonForAppointment}
-        routeToNextAppointmentPage={routeToNextAppointmentPage}
-        data={{
-          reasonForAppointment: 'routine-follow-up',
-          reasonAdditionalInfo: 'test',
-        }}
-      />,
+    expect(screen.baseElement).to.contain.text(
+      'Tell us the reason for this appointment',
     );
-    form.find('form').simulate('submit');
 
-    expect(form.find('.usa-input-error').length).to.equal(0);
+    const textBox = screen.getByRole('textbox');
+    fireEvent.change(textBox, { target: { value: '   ' } });
+    expect(textBox.value).to.equal('   ');
+    fireEvent.click(screen.getByText(/Continue/));
 
-    form.unmount();
+    expect(await screen.findByRole('alert')).to.contain.text(
+      'Please provide a response',
+    );
   });
 
-  it('should render alert message', () => {
-    const openReasonForAppointment = sinon.spy();
-    const updateReasonForAppointmentData = sinon.spy();
-
-    const form = mount(
-      <ReasonForAppointmentPage
-        openReasonForAppointment={openReasonForAppointment}
-        updateReasonForAppointmentData={updateReasonForAppointmentData}
-        data={{}}
-      />,
+  it('should continue to the correct page based on type choice for VA medical request', async () => {
+    const store = createTestStore(initialState);
+    const screen = renderWithStoreAndRouter(
+      <Route component={ReasonForAppointmentPage} />,
+      {
+        store,
+      },
     );
-    expect(form.text()).to.contain(
-      'If you have an urgent medical need, please',
+
+    fireEvent.click(
+      await screen.findByLabelText(/Routine or follow-up visit/i),
     );
+    const textBox = screen.getByRole('textbox');
+    fireEvent.change(textBox, { target: { value: 'test' } });
+    expect(textBox.value).to.equal('test');
 
-    form.unmount();
-  });
+    fireEvent.click(screen.getByText(/Continue/));
 
-  it('document title should match h1 text', () => {
-    const openReasonForAppointment = sinon.spy();
-    const updateReasonForAppointmentData = sinon.spy();
-    const pageTitle = 'Choose a reason for your appointment';
+    await waitFor(() =>
+      expect(screen.history.push.lastCall?.args[0]).to.equal(
+        '/new-appointment/choose-visit-type',
+      ),
+    );
 
-    const form = mount(
-      <ReasonForAppointmentPage
-        openReasonForAppointment={openReasonForAppointment}
-        updateReasonForAppointmentData={updateReasonForAppointmentData}
-        data={{}}
-      />,
+    fireEvent.click(
```

Comment:
The logic for choosing the next page after this one isn't conditional on the options, so I don't think you need to test that each option takes you to the right next page.

Reply:
Understood